### DISCUSSION TITLE
Stable Scroll Restoration functionality

### DISCUSSION THREAD

**Discussion by vinny17690** [vercel/next.js#34906] [Ideas]
### Describe the feature you'd like to request

Stable `scrollRestoration` functionality

### Describe the solution you'd like

The current experimental features are working fine for us, but we are hesitant to use it on the production. Wondering what's ETA for moving `scrollRestoration` functionality to stable core? Also, is there any plan to document it further more? The current documentation is not so great.

### Describe alternatives you've considered

Experimental feature

**Comment by paales** [comment]
Yes! I want this as well! We do some animation magic when navigating pages and we need to know the scroll position of the pages to properly handle this. We tried to roll our own, but there isn't a proper place to handle the timing correctly and the scrollRestoration does exactly what we want.
